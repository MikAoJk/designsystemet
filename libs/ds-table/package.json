{
  "name": "@skatteetaten/ds-table",
  "version": "0.0.1",
  "main": "./index.js",
  "module": "./index.mjs",
  "groupId": "no.skatteetaten.aurora",
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "react": "^18"
  }
}
