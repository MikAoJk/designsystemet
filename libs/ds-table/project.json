{
  "name": "ds-table",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-table/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-table/**/*.{ts,tsx,js,jsx}",
          "libs/ds-table/package.json"
        ]
      }
    },
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-table",
        "tsConfig": "libs/ds-table/tsconfig.lib.json",
        "project": "libs/ds-table/package.json",
        "entryFile": "libs/ds-table/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-table/rollup.config.js"],
        "compiler": "babel",
        "assets": [
          {
            "glob": "libs/ds-table/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-table/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-table/**/*.css",
          "libs/ds-table/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      }
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-table"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-table {args.switches}",
        "cwd": ""
      }
    }
  },
  "implicitDependencies": ["!ds-dev-config"]
}
