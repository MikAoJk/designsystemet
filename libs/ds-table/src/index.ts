export * from './Table/Table';
export * from './Table/Table.types';
