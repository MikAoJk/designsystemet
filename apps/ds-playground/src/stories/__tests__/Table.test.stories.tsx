import { Table, TableProps } from '@skatteetaten/ds-table';
import { expect } from '@storybook/jest';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { within } from '@storybook/testing-library';

const verifyAttribute =
  (attribute: string, expectedValue: string) =>
  async ({ canvasElement }: { canvasElement: HTMLElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const table = canvas.getByRole('table');
    await expect(table).toBeInTheDocument();
    await expect(table).toHaveAttribute(attribute, expectedValue);
  };

export default {
  component: Table,
  title: 'Tester/Table',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    caption: { table: { disable: true } },
    variant: { table: { disable: true } },
    hasFullWidth: { table: { disable: true } },
    showCaption: { table: { disable: true } },
    sortState: { table: { disable: true } },
    setSortState: { table: { disable: true } },
    children: { table: { disable: true } },
  },
} as ComponentMeta<typeof Table>;

const defaultArgs: TableProps = {
  caption: 'tabellcaption',
};

const Template: ComponentStory<typeof Table> = (args) => (
  <Table {...args} variant={args.variant}>
    <Table.Header>
      <Table.Row>
        <Table.HeaderCell scope={'col'}>{'Items'}</Table.HeaderCell>
        <Table.HeaderCell scope={'col'}>{'Expenditure'}</Table.HeaderCell>
        <Table.HeaderCell scope={'col'}>{'Category'}</Table.HeaderCell>
      </Table.Row>
    </Table.Header>
    <Table.Body>
      <Table.Row>
        <Table.DataCell as={'th'}>{'Donuts'}</Table.DataCell>
        <Table.DataCell>{'3,000'}</Table.DataCell>
        <Table.DataCell className={'rowSpanLeft'} rowSpan={2}>
          {'Edible'}
        </Table.DataCell>
      </Table.Row>
      <Table.Row>
        <Table.DataCell as={'th'}>{'Cake'}</Table.DataCell>
        <Table.DataCell>{'3,000'}</Table.DataCell>
      </Table.Row>
      <Table.Row>
        <Table.DataCell as={'th'}>{'Stationery'}</Table.DataCell>
        <Table.DataCell>{'18,000'}</Table.DataCell>
        <Table.DataCell>{'Non-Edible'}</Table.DataCell>
      </Table.Row>
    </Table.Body>
  </Table>
);

// Når Table har en ref, så får dom table elementet ref forwarded
export const WithRef = Template.bind({});
WithRef.storyName = 'With Ref (FA1)';
WithRef.args = {
  ...defaultArgs,
  ref: (instance: HTMLTableElement | null): void => {
    if (instance) {
      instance.id = 'dummyIdForwardedFromRef';
    }
  },
};
WithRef.argTypes = {
  ref: { table: { disable: false } },
};
WithRef.parameters = {
  imageSnapshot: { disable: true },
};
WithRef.play = verifyAttribute('id', 'dummyIdForwardedFromRef');

// Når IconButton har en id, så har a-element id
// Når IconButton har en custom CSS, så vises custom stil
// Når IconButton har en lang, så har a-element lang
// Når IconButton har dataTestid, så har a-elementet data-testid satt
export const WithAttributes = Template.bind({});
WithAttributes.storyName = 'With Attributes(FA2-5)';
WithAttributes.args = {
  ...defaultArgs,
  id: 'htmlId',
  className: 'dummyClassname',
  lang: 'nb',
  'data-testid': '123ID',
};
WithAttributes.argTypes = {
  ...WithAttributes.argTypes,
  id: { table: { disable: false } },
  className: { table: { disable: false } },
  lang: { table: { disable: false } },
  'data-testid': { table: { disable: false } },
};
WithAttributes.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const table = canvas.getByRole('table');
  await expect(table).toHaveClass('dummyClassname');
  await expect(table).toHaveAttribute('id', 'htmlId');
  await expect(table).toHaveAttribute('lang', 'nb');
  await expect(table).toHaveAttribute('data-testid', '123ID');
};

export const WithVariantCompact = Template.bind({});
WithVariantCompact.storyName = 'Variant compact(A1)';
WithVariantCompact.args = {
  ...defaultArgs,
  variant: 'compact',
};
WithVariantCompact.argTypes = {
  variant: { table: { disable: false } },
};

const TemplateScroll: ComponentStory<typeof Table> = (args) => {
  const exampleTable = (
    <Table {...args} variant={args.variant}>
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell scope={'col'}>{'Forename'}</Table.HeaderCell>
          <Table.HeaderCell scope={'col'}>{'Surname'}</Table.HeaderCell>
          <Table.HeaderCell scope={'col'}>{'email'}</Table.HeaderCell>
        </Table.Row>
      </Table.Header>
      <Table.Body>
        <Table.Row>
          <Table.DataCell>{'Otto'}</Table.DataCell>
          <Table.DataCell>{'Octavius'}</Table.DataCell>
          <Table.DataCell>{'doc.ock@example.org'}</Table.DataCell>
        </Table.Row>
        <Table.Row>
          <Table.DataCell>{'Norman'}</Table.DataCell>
          <Table.DataCell>{'Osbourne'}</Table.DataCell>
          <Table.DataCell>{'norman.osbourne@example.org'}</Table.DataCell>
        </Table.Row>
        <Table.Row>
          <Table.DataCell>{'Curt'}</Table.DataCell>
          <Table.DataCell>{'Connors'}</Table.DataCell>
          <Table.DataCell>{'curt.connors@example.org'}</Table.DataCell>
        </Table.Row>
      </Table.Body>
    </Table>
  );

  return (
    <div>
      <p>{'Table med Scrollbar'}</p>
      <div className={'scrollTableWrapper'}>{exampleTable}</div>
      <div className={'scrollTableWrapperSuccess'}>{exampleTable}</div>
      <div className={'scrollTableWrapperWarning'}>{exampleTable}</div>
    </div>
  );
};

// Når Table får en scrollbar så tegnes den opp som forventet
export const WithScrollbar = TemplateScroll.bind({});
WithScrollbar.storyName = 'With scroll(A5)';
WithScrollbar.parameters = { a11y: { disable: true } };
