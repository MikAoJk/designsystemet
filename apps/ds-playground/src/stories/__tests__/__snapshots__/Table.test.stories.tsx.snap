// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Table WithAttributes play-test 1`] = `
<div class=" ">
  <table id="htmlId"
         class="Table_table__VhSJI  Table_table_standard__8JWMb dummyClassname"
         lang="nb"
         data-testid="123ID"
  >
    <caption class="Table_tableCaption__zTm5t Table_hidden__N7qKc Table_tableCaption_standard__Nhqht">
      tabellcaption
    </caption>
    <thead>
      <tr class="Row_row_notExpanded__qKGMb">
        <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
            scope="col"
        >
          Items
        </th>
        <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
            scope="col"
        >
          Expenditure
        </th>
        <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
            scope="col"
        >
          Category
        </th>
      </tr>
    </thead>
    <tbody>
      <tr class="Row_row_notExpanded__qKGMb">
        <th class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  "
            scope="row"
        >
          Donuts
        </th>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
          3,000
        </td>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC DataCell_dataCell_rowspan__xTjXX rowSpanLeft"
            rowspan="2"
        >
          Edible
        </td>
      </tr>
      <tr class="Row_row_notExpanded__qKGMb">
        <th class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  "
            scope="row"
        >
          Cake
        </th>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
          3,000
        </td>
      </tr>
      <tr class="Row_row_notExpanded__qKGMb">
        <th class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  "
            scope="row"
        >
          Stationery
        </th>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
          18,000
        </td>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
          Non-Edible
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Tester/Table WithRef play-test 1`] = `
<div class=" ">
  <table class="Table_table__VhSJI  Table_table_standard__8JWMb "
         id="dummyIdForwardedFromRef"
  >
    <caption class="Table_tableCaption__zTm5t Table_hidden__N7qKc Table_tableCaption_standard__Nhqht">
      tabellcaption
    </caption>
    <thead>
      <tr class="Row_row_notExpanded__qKGMb">
        <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
            scope="col"
        >
          Items
        </th>
        <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
            scope="col"
        >
          Expenditure
        </th>
        <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
            scope="col"
        >
          Category
        </th>
      </tr>
    </thead>
    <tbody>
      <tr class="Row_row_notExpanded__qKGMb">
        <th class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  "
            scope="row"
        >
          Donuts
        </th>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
          3,000
        </td>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC DataCell_dataCell_rowspan__xTjXX rowSpanLeft"
            rowspan="2"
        >
          Edible
        </td>
      </tr>
      <tr class="Row_row_notExpanded__qKGMb">
        <th class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  "
            scope="row"
        >
          Cake
        </th>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
          3,000
        </td>
      </tr>
      <tr class="Row_row_notExpanded__qKGMb">
        <th class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  "
            scope="row"
        >
          Stationery
        </th>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
          18,000
        </td>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
          Non-Edible
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Tester/Table WithScrollbar smoke-test 1`] = `
<div>
  <p>
    Table med Scrollbar
  </p>
  <div class="scrollTableWrapper">
    <div class="Table_wrapper__cQelY ">
      <table class="Table_table__VhSJI  Table_table_standard__8JWMb ">
        <thead>
          <tr class="Row_row_notExpanded__qKGMb">
            <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
                scope="col"
            >
              Forename
            </th>
            <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
                scope="col"
            >
              Surname
            </th>
            <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
                scope="col"
            >
              email
            </th>
          </tr>
        </thead>
        <tbody>
          <tr class="Row_row_notExpanded__qKGMb">
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Otto
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Octavius
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              doc.ock@example.org
            </td>
          </tr>
          <tr class="Row_row_notExpanded__qKGMb">
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Norman
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Osbourne
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              norman.osbourne@example.org
            </td>
          </tr>
          <tr class="Row_row_notExpanded__qKGMb">
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Curt
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Connors
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              curt.connors@example.org
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div class="scrollTableWrapperSuccess">
    <div class="Table_wrapper__cQelY ">
      <table class="Table_table__VhSJI  Table_table_standard__8JWMb ">
        <thead>
          <tr class="Row_row_notExpanded__qKGMb">
            <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
                scope="col"
            >
              Forename
            </th>
            <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
                scope="col"
            >
              Surname
            </th>
            <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
                scope="col"
            >
              email
            </th>
          </tr>
        </thead>
        <tbody>
          <tr class="Row_row_notExpanded__qKGMb">
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Otto
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Octavius
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              doc.ock@example.org
            </td>
          </tr>
          <tr class="Row_row_notExpanded__qKGMb">
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Norman
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Osbourne
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              norman.osbourne@example.org
            </td>
          </tr>
          <tr class="Row_row_notExpanded__qKGMb">
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Curt
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Connors
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              curt.connors@example.org
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div class="scrollTableWrapperWarning">
    <div class="Table_wrapper__cQelY ">
      <table class="Table_table__VhSJI  Table_table_standard__8JWMb ">
        <thead>
          <tr class="Row_row_notExpanded__qKGMb">
            <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
                scope="col"
            >
              Forename
            </th>
            <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
                scope="col"
            >
              Surname
            </th>
            <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_standard__SWZS8  HeaderCell_headerCell_left__KiiuU undefined"
                scope="col"
            >
              email
            </th>
          </tr>
        </thead>
        <tbody>
          <tr class="Row_row_notExpanded__qKGMb">
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Otto
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Octavius
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              doc.ock@example.org
            </td>
          </tr>
          <tr class="Row_row_notExpanded__qKGMb">
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Norman
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Osbourne
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              norman.osbourne@example.org
            </td>
          </tr>
          <tr class="Row_row_notExpanded__qKGMb">
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Curt
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              Connors
            </td>
            <td class="DataCell_dataCell__t3nsq DataCell_dataCell_standard__uNRyN DataCell_dataCell_left__KhBFC  ">
              curt.connors@example.org
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Tester/Table WithVariantCompact smoke-test 1`] = `
<div class=" ">
  <table class="Table_table__VhSJI  Table_table_compact__MEM6O ">
    <caption class="Table_tableCaption__zTm5t Table_hidden__N7qKc Table_tableCaption_compact__AmeMW">
      tabellcaption
    </caption>
    <thead>
      <tr class="Row_row_notExpanded__qKGMb">
        <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_compact__tyAcv  HeaderCell_headerCell_left__KiiuU undefined"
            scope="col"
        >
          Items
        </th>
        <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_compact__tyAcv  HeaderCell_headerCell_left__KiiuU undefined"
            scope="col"
        >
          Expenditure
        </th>
        <th class="HeaderCell_headerCell__-PNsO HeaderCell_headerCell_compact__tyAcv  HeaderCell_headerCell_left__KiiuU undefined"
            scope="col"
        >
          Category
        </th>
      </tr>
    </thead>
    <tbody>
      <tr class="Row_row_notExpanded__qKGMb">
        <th class="DataCell_dataCell__t3nsq DataCell_dataCell_compact__POiKz DataCell_dataCell_left__KhBFC  "
            scope="row"
        >
          Donuts
        </th>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_compact__POiKz DataCell_dataCell_left__KhBFC  ">
          3,000
        </td>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_compact__POiKz DataCell_dataCell_left__KhBFC DataCell_dataCell_rowspan__xTjXX rowSpanLeft"
            rowspan="2"
        >
          Edible
        </td>
      </tr>
      <tr class="Row_row_notExpanded__qKGMb">
        <th class="DataCell_dataCell__t3nsq DataCell_dataCell_compact__POiKz DataCell_dataCell_left__KhBFC  "
            scope="row"
        >
          Cake
        </th>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_compact__POiKz DataCell_dataCell_left__KhBFC  ">
          3,000
        </td>
      </tr>
      <tr class="Row_row_notExpanded__qKGMb">
        <th class="DataCell_dataCell__t3nsq DataCell_dataCell_compact__POiKz DataCell_dataCell_left__KhBFC  "
            scope="row"
        >
          Stationery
        </th>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_compact__POiKz DataCell_dataCell_left__KhBFC  ">
          18,000
        </td>
        <td class="DataCell_dataCell__t3nsq DataCell_dataCell_compact__POiKz DataCell_dataCell_left__KhBFC  ">
          Non-Edible
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
